numbat/tests/interpreter.rs

```diff
@@ -43,6 +48,48 @@ fn expect_exact_failure(code: &str, expected: &str) {
     }
 }
 
+#[test]
+fn simple_value() {
```

Comment:
Do these tests need to be full-on integration tests? These `expect_output` calls are quite heavy as we load/compile/interpret the full prelude for each of them.

I'm okay with leaving them here if it's too much work to convert them to tokenizer or parser unit tests.

Reply:
Oh, I didn't think about the perf.

> Do these tests need to be full-on integration tests? Also, in which way do they differ from the tokenizer unit test?

Usually, in meilisearch I duplicate the lexer/parser tests as full integration tests just to be sure that no step in the middle broke or hide a behaviour (in the parser or interpreter for example)

> These expect_output calls are quite heavy as we load/compile/interpret the full prelude for each of them.

Sooo, if I understand correctly, the last solution could be to use the `expect_output_with_context` and share the same context between each test?

Reply:
> Usually, in meilisearch I duplicate the lexer/parser tests as full integration tests going through the whole HTTP server just to be sure that no step in the middle broke or hide a behavior (in the parser or interpreter, for example). We’ve had so many dumb issues like that in the past by only testing the parser without the interpreter.

I can certainly sympathize :smile:. I usually try to have the fine-grained tests in the unit tests / the module itself. And then one generic example in the interpreter/integration tests.

> Sooo, if I understand correctly, another solution could be to use the `expect_output_with_context` and share the same context between each test?

Yes, but the context is mutable. So we don't want to do this by default. In principle, we want a clean environment for each test. What we *could* probably do is to generate the initial context once, and then clone it? That might be a nice speedup for the unit tests.

Reply:
So I think we can merge this as-is, unless you want to add something in this PR?

Reply:
I wanted to update the tests, as we said, but didn't really find the time to do so 😅 
If you're ok to wait, I’ll probably find the time to do it this week or next week

Reply:
Of course, there is no rush.